core/src/main/java/org/infinispan/configuration/cache/AbstractStoreConfigurationBuilder.java

```diff
@@ -239,9 +239,14 @@ private void validateStoreAttributes() {
          throw log.attributeNotAllowedInInvalidationMode(FETCH_PERSISTENT_STATE.name());
       }
 
-      if (shared && !preload && builder.indexing().enabled()
-            && builder.indexing().indexLocalOnly() && !getBuilder().template())
-         log.localIndexingWithSharedCacheLoaderRequiresPreload();
+      if (shared) {
+         if (!builder.clustering().cacheMode().isClustered())
+            throw log.sharedStoreWithLocalCache();
+
+         if (!preload && builder.indexing().enabled()
+               && builder.indexing().indexLocalOnly() && !getBuilder().template())
```

Comment:
```indexLocalOnly``` is deprecated. Would be possible to change to the new way to test this in this PR?

Reply:
Tbh I don't know enough about Indexing. From what I can see this configuration combo is still invalid and `indexLocalOnly` is still used by the `IndexModificationStrategy`, so I think we still need this check.

Reply:
ok!